### ISSUE TITLE
No warning for mismatched nullability in constraint

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#27742]
No warning reported for `new B<A<object>>()`:
```c#
class A<T> { }
class B<T> where T : A<object?> { }
class C
{
    static void Main()
    {
        new B<A<object?>>();
        new B<A<object>>(); // warning
    }
}
```

**Comment by cston** [comment]
Fixed in #28959.
